controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -352,52 +353,20 @@ public void getDelegationToken(io.pravega.controller.stream.api.grpc.v1.Controll
                             streamObserver.onCompleted();
                         }
                     });
-        } else {
+        } catch (Exception e) {
             log.error("Controller api failed with authenticator error");
             streamObserver.onError(Status.UNAUTHENTICATED
                     .withDescription("Authentication failed")
                     .asRuntimeException());
         }
     }
 
-    public boolean checkAuthorization(String resource, AuthHandler.Permissions expectedLevel) {
-        if (isAuthEnabled) {
-            PravegaInterceptor currentInterceptor = PravegaInterceptor.INTERCEPTOR_OBJECT.get();
-
-            AuthHandler.Permissions allowedLevel;
-            if (currentInterceptor == null) {
-                //No interceptor, and authorization is enabled. Means no access is granted.
-                allowedLevel = AuthHandler.Permissions.NONE;
-            } else {
-                allowedLevel = currentInterceptor.authorize(resource);
-            }
-            if (allowedLevel.ordinal() < expectedLevel.ordinal()) {
-                return false;
-            }
-            return true;
-        } else {
-            return true;
-        }
+    private String checkAuthorization(String resource, AuthHandler.Permissions expectedLevel) {
+        return this.authHelper.checkAuthorization(resource, expectedLevel);
```

Comment:
These two could probably be inlined, as they are private.

Reply:
Inlined the calls.